llama-index-integrations/llms/llama-index-llms-mistral-rs/llama_index/llms/mistral_rs/base.py

```diff
@@ -0,0 +1,302 @@
+        top_k: int = DEFAULT_TOPK,
+        top_p: int = DEFAULT_TOPP,
+        top_logprobs: Optional[int] = DEFAULT_TOP_LOGPROBS,
+        callback_manager: Optional[CallbackManager] = None,
+        generate_kwargs: Optional[Dict[str, Any]] = None,
+        model_kwargs: Optional[Dict[str, Any]] = {},
+        system_prompt: Optional[str] = None,
+        messages_to_prompt: Optional[Callable[[Sequence[ChatMessage]], str]] = None,
+        completion_to_prompt: Optional[Callable[[str], str]] = None,
+        pydantic_program_mode: PydanticProgramMode = PydanticProgramMode.DEFAULT,
+        output_parser: Optional[BaseOutputParser] = None,
+    ) -> None:
+        generate_kwargs = generate_kwargs or {}
+        generate_kwargs.update(
+            {
+                "temperature": temperature,
+                "max_tokens": max_new_tokens,
+                "top_k": top_k,
+                "top_p": top_p,
+                "top_logprobs": top_logprobs,
+            }
+        )
+
+        super().__init__(
+            model_path="local",
+            model_url="local",
+            temperature=temperature,
+            context_window=context_window,
+            max_new_tokens=max_new_tokens,
+            callback_manager=callback_manager,
+            generate_kwargs=generate_kwargs,
+            model_kwargs={},
+            verbose=True,
+            system_prompt=system_prompt,
+            messages_to_prompt=messages_to_prompt,
+            completion_to_prompt=completion_to_prompt,
+            pydantic_program_mode=pydantic_program_mode,
+            output_parser=output_parser,
+        )
+
+        self._runner = Runner(
+            which=which,
+            token_source=model_kwargs.get("token_source", "cache"),
+            max_seqs=model_kwargs.get("max_seqs", DEFAULT_MAX_SEQS),
+            prefix_cache_n=model_kwargs.get("prefix_cache_n", DEFAULT_PREFIX_CACHE_N),
+            no_kv_cache=model_kwargs.get("no_kv_cache", False),
+            chat_template=model_kwargs.get("chat_template", None),
+        )
+        self._has_messages_to_prompt = messages_to_prompt is not None
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "MistralRS_llm"
+
+    @property
+    def metadata(self) -> LLMMetadata:
+        """LLM metadata."""
+        return LLMMetadata(
+            context_window=self.context_window,
+            num_output=self.max_new_tokens,
+            model_name=self.model_path,
+        )
+
+    @llm_chat_callback()
+    def chat(self, messages: Sequence[ChatMessage], **kwargs: Any) -> ChatResponse:
+        if self._has_messages_to_prompt:
+            messages = self.messages_to_prompt(messages)
+        else:
+            messages = llama_index_to_mistralrs_messages(messages)
+        self.generate_kwargs.update({"stream": False})
+
+        request = ChatCompletionRequest(
+            messages=messages,
+            model="",
+            logit_bias=None,
+            logprobs=False,
```

Comment:
does this support LogProbs? It would be nice to add them here if they did.

Reply:
Sorry, re-opening this as I don't think we are including the `LogProb` in the `CompletionResponse`.

https://github.com/run-llama/llama_index/blob/802064aee72b03ab38ead0cda780cfa3e37ce728/llama-index-core/llama_index/core/base/llms/types.py#L120

and in your `base.py` on L. 270 you have:

```python
return CompletionResponse(text=completion_response.choices[0].message.content)
```

Reply:
Sorry, I didn't notice that. I've added that now.